### ISSUE TITLE
[BUG]: beanie cannot be imported in Python 3.11 using current MongoDB's motor

### ISSUE THREAD

**Issue by mikeckennedy** [roman-right/beanie#389]
**Describe the bug**
Should be able to "import beanie"

**To Reproduce**
Just run the line: `import beanie` in Python 3.11

**Expected behavior**
Runs successfully.

**Additional context**
Get this error, just try to import in the REPL

```
Python 3.11.0 (main, Oct 24 2022, 19:46:22) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import beanie
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/code/venv/lib/python3.11/site-packages/beanie/__init__.py", line 1, in <module>
    from beanie.migrations.controllers.free_fall import free_fall_migration
  File "/code/venv/lib/python3.11/site-packages/beanie/migrations/controllers/free_fall.py", line 4, in <module>
    from beanie.odm.documents import Document
  File "/code/venv/lib/python3.11/site-packages/beanie/odm/documents.py", line 19, in <module>
    from motor.motor_asyncio import AsyncIOMotorDatabase
  File "/code/venv/lib/python3.11/site-packages/motor/motor_asyncio.py", line 18, in <module>
    from .frameworks import asyncio as asyncio_framework
  File "/code/venv/lib/python3.11/site-packages/motor/frameworks/asyncio/__init__.py", line 27, in <module>
    from asyncio import coroutine  # noqa: F401 - For framework interface.
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name 'coroutine' from 'asyncio' (/usr/local/lib/python3.11/asyncio/__init__.py)
```

**Comment by mikeckennedy** [comment]
@roman-right I put this here so you know, but realize it may a knock on error from motor. :(

**Comment by mikeckennedy** [comment]
It looks like `asyncio.coroutine` was removed intentionally as a breaking change:

https://github.com/python/cpython/issues/87382

Sadly, it seems MongoDB hasn't been on top of that. I checked the github repo for motor and they don't have public issues section so I can't file it there. 

@roman-right , feel free to close this now or we can wait until the next release of `motor` which hopefully would fix this.
